lib/iris/util.py

```diff
@@ -1517,6 +1517,111 @@ def demote_dim_coord_to_aux_coord(cube, name_or_coord):
     cube.add_aux_coord(dim_coord, coord_dim)
 
 
+def flatten_multidim_coord(cube, name_or_coord):
+    """
+    Flatten the Aux coord and associated dimensions on the cube
+
+
+    * cube
+        An instance of :class:`iris.cube.Cube`
+
+    * name_or_coord:
+        Either
+
+        (a) An instance of :class:`iris.coords.DimCoord`
+
+        or
+
+        (b) the :attr:`standard_name`, :attr:`long_name`, or
+        :attr:`var_name` of an instance of an instance of
+        :class:`iris.coords.DimCoord`.
+
+    For example::
+
+        >>> print(cube)
+        rainfall_rate / (kg m-2 s-1)        (x: 5; y: 3)
+             Dimension coordinates:
+                  x                           x     -
+                  y                           -     x
+             Auxiliary coordinates:
+                  latitude                    x     x
+                  longitude                   x     x
+
+        >>> flatten_multidim_coord(cube, 'longitude')
+        >>> print(cube)
+        rainfall_rate / (kg m-2 s-1)        (--: 15)
+             Auxiliary coordinates:
+                  latitude                    x
+                  longitude                   x
+    """
+    import numpy as np
+    from iris.cube import Cube
+
+    if isinstance(name_or_coord, six.string_types):
+        coord = cube.coord(name_or_coord)
+    elif isinstance(name_or_coord, iris.coords.Coord):
+        coord = name_or_coord
+    else:
+        # Don't know how to handle this type
+        msg = ("Don't know how to handle coordinate of type {}. "
+               "Ensure all coordinates are of type six.string_types or "
+               "iris.coords.Coord.")
+        msg = msg.format(type(name_or_coord))
+        raise TypeError(msg)
+
+    coord_dims = cube.coord_dims(coord)
+
+    if len(coord_dims) == 1:
+        # Do nothing
+        return cube
+
+    other_dims = list(set(range(cube.ndim)) - set(coord_dims))
+    shape_array = np.asarray(cube.shape)
+    new_shape = np.hstack([shape_array[other_dims],
+                           np.product(shape_array[list(coord_dims)])])
+
+    # Get all the coords which span the dimensions to be flattened
+    coords_to_flatten = cube.coords(dimensions=coord_dims)
+    # And any other coords which span any one of the dimensions
+    # These (1-D) coordinates are expanded before being flattened
+    coords_to_expand = []
+    for d in coord_dims:
+        coords_to_expand.extend(cube.coords(dimensions=d))
+
+    # Expand the coords which need expanding first, then add to flatten list
+    for c in coords_to_expand:
+        new_points = broadcast_to_shape(c.points, coord.shape, (0,))
+        new_bounds = broadcast_to_shape(c.bounds, coord.bounds.shape, (0,)) \
+            if c.bounds is not None else None
+        coords_to_flatten.append(c.copy(new_points, new_bounds))
```

Comment:
Test is failing here because its trying to create a 2d DimCoord.  You can get around it with
```python
coords_to_flatten.append(
    iris.coords.AuxCoord.from_coord(c).copy(new_points, new_bounds))
```
though there may be a better way.

Then the test **still** fails when the cube is constructed because it claims the new 'blah' is only length 3.  I do not understand this at all.

Reply:
Yes, good catch thanks - I've fixed that now.